projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row-dimension-content.component.ts

```diff
@@ -175,18 +180,26 @@ export class IgxPivotRowDimensionContentComponent extends IgxGridHeaderRowCompon
     }
 
     protected extractFromDimension(dimData, rowData: any[], lvl) {
-        const field = dimData.dimension.memberName;
-        const header = rowData[field];
-        const col = this._createColComponent(field, header, dimData.dimension, lvl);
+        const field = dimData?.dimension?.memberName || 'placeholder';
+        const header = rowData[field] || '';
+        const col = this._createColComponent(field, header, dimData?.dimension, lvl);
         return col;
     }
 
     protected _createColComponent(field: string, header: string, dim: IPivotDimension, lvl) {
         const ref = this.viewRef.createComponent(IgxColumnComponent);
         ref.instance.field = field;
         ref.instance.header = header;
-        ref.instance.width = this.grid.resolveRowDimensionWidth(this.dimension) + 'px';
         ref.instance.resizable = this.grid.rowDimensionResizing;
+
+        if (!dim) {
```

Comment:
With the latest changes, there should always be a dimension, so maybe the all changes in this file that check for undefined dimension are no longer necessary.